Cinema system/Cinema system/Services/CinemaService.cs

```diff
@@ -81,5 +83,105 @@ public void DeleteCinema(int cinemaId)
 
             _db.SaveChanges();
         }
+
+        public bool IsHallExists(int cinemaId, int hallId)
+        {
+            return _db.Halls.Any(h => (h.Id == hallId && h.CinemaId == cinemaId && !h.IsDeleted));
+        }
+
+        public bool IsHallExists(int cinemaId, HallView hall)
+        {
+            return _db.Halls.Any(h => (h.Name == hall.Name && h.CinemaId == cinemaId && !h.IsDeleted));
+        }
+
+        public int CreateHall(int cinemaId, HallView hall)
+        {
+            var newHall = new Hall
+            {
+                Id = hall.Id,
+                Name = hall.Name,
+                Seats = hall.Seats
+                    .Select(s => new Seat
+                    {
+                        TypeOfSeat = (SeatType)Enum.Parse(typeof(SeatType), s.TypeOfSeat),
+                        Row = s.Row,
+                        Place = s.Place
+                    }).ToList()
+            };
+
+            var cinema = _db.Cinemas
+                 .Include(c => c.Halls)
+                 .FirstOrDefault(c => c.Id == cinemaId);
+
+            cinema.Halls.Add(newHall);
+
+            _db.SaveChanges();
+
+            return newHall.Id;
+        }
+
+        public void EditHall(int hallId, HallView hall)
+        {
+            var editHall = _db.Halls
+                .Include(h => h.Seats)
+                .FirstOrDefault(h => h.Id == hallId);
```

Comment:
Why first or default? Are you expecting multiple records with the same Id in database?